internal/pkg/cli/svc_init.go

```diff
@@ -223,15 +226,18 @@ func (o *initSvcOpts) Execute() error {
 		}
 	}
 
-	platform, err := o.dockerEngine.RedirectPlatform(o.image)
+	detectedPlatform, targetPlatform, err := o.dockerEngine.RedirectPlatform(o.image)
 	if err != nil {
 		return fmt.Errorf("get/redirect docker engine platform: %w", err)
 	}
-	o.platform = platform
+	o.platform = targetPlatform
 	if o.platform != nil {
-		log.Warningf(`Your architecture type is currently unsupported. Setting platform %s instead.\n`, dockerengine.PlatformString(dockerengine.OSLinux, dockerengine.ArchAMD64))
-		if o.wkldType != manifest.RequestDrivenWebServiceType {
-			log.Warning("See 'platform' field in your manifest.\n")
+		// Error out early if Windows && AppRunner/RDWS.
+		if strings.Split(aws.StringValue(targetPlatform), "/")[0] == manifest.OSWindows && o.wkldType == manifest.RequestDrivenWebServiceType {
```

Comment:
can we define `targetPlatform` to it's own type
```go
type Platform string

func (p *Platform) Parse() (string, string)
```
something like this would increase both readability and reliability...